sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -574,6 +574,15 @@ case class DeleteFromTable(
   override def children: Seq[LogicalPlan] = child :: Nil
 }
 
+case class UpdateTable(
+    child: LogicalPlan,
+    attrs: Seq[Attribute],
```

Comment:
can we really use `Seq[Attribute]`? When Spark resolves it to nested field, it will be `Alias` which is not an `Attribute`, and we will get weird errors.

Reply:
Metioned in https://github.com/apache/spark/pull/25626#discussion_r320151744, `Seq[Attribute]` involves the ugly `.asInstanceOf[Seq[Any]]` when checking the type of fields. But need we update it to `Seq[Expression]`? My feeling is the latter is too general to represent a field.

Reply:
how about `Seq[NamedExpression]`?

Reply:
`Seq[NamedExpression]` is ok for me, and I updated the code. But `Seq[NamedExpression]` is not a perfect solution, as some nested field does not have a name, which would be resolved to something like `GetStructField`, and it is a subclass of `Expression`.
Updating a nested field is complex. For a named field user can update it by specifying the name, but for an array, or a more complex json which composed by nested objects and arrays, it's hard to specify the field to be updated.
So currrently, my think is we can limit the updating to the scope of non-nested field, which can be resolved to a `NamedExpression`. What's your opnion? @cloud-fan

Reply:
Then I think we have no choice but use `Seq[Expression]`. We should add comments to explain that `attrs` here holds the expressions that maybe attributes.

Reply:
Or we can change the analyzer to only resolve column names in `UpdateTable` as attributes, which I don't think is worthwhile.

Reply:
Updated it to `Seq[Expression]` and added some explaining doc, pls review.